packages/common/src/services/auth/hedgehog.ts

```diff
@@ -0,0 +1,93 @@
+import { Hedgehog, WalletManager, getPlatformCreateKey } from '@audius/hedgehog'
+import type { SetAuthFn, SetUserFn, GetFn, CreateKey } from '@audius/hedgehog'
+
+import type { LocalStorage } from '../local-storage'
+
+import type { IdentityService } from './identity'
+
+export type HedgehogConfig = {
+  identityService: IdentityService
+  useLocalStorage?: boolean
+  localStorage?: LocalStorage
+  createKey?: CreateKey
+}
+
+export type HedgehogInstance = Hedgehog & {
```

Comment:
I discarded the class pattern we were using before for two reasons:
1. Getting the type correct was difficult in that pattern since we weren't actually _extending_ the base hedgehog class, just decorating it and then playing around with `this` pointers 😬 
2. The above instance shenanigans are hard to grok and feel error-prone.

Reply:
We do still need those because the type on the class doesn't include those methods so TS gets mad about us extending the object. I can fiddle with it though and see if I can get it to work with casting.